packages/typespec-ts/src/utils/operationUtil.ts

```diff
@@ -44,14 +45,85 @@ import {
 import { isByteOrByteUnion } from "./modelUtils.js";
 import { SdkContext } from "./interfaces.js";
 
+// Sorts the responses by status code
+export function sortedOperationResponses(responses: HttpOperationResponse[]) {
+  return responses.sort((a, b) => {
+    if (a.statusCodes === "*") {
+      return 1;
+    }
+    if (b.statusCodes === "*") {
+      return -1;
+    }
+    const aStatus =
+      typeof a.statusCodes === "number" ? a.statusCodes : a.statusCodes.start;
+    const bStatus =
+      typeof b.statusCodes === "number" ? b.statusCodes : b.statusCodes.start;
+    return aStatus - bStatus;
+  });
+}
+
+/**
+ * This function computes all the response types error and success
+ * an operation can end up returning.
+ */
+export function getOperationResponseTypes(
+  dpgContext: SdkContext,
+  operation: HttpOperation
+): ResponseTypes {
+  const returnTypes: ResponseTypes = {
+    error: [],
+    success: []
+  };
+  function getResponseType(responses: HttpOperationResponse[]) {
+    return responses.map((r) => {
+      const statusCode = getOperationStatuscode(r);
+      const responseName = getResponseTypeName(
+        getOperationGroupName(dpgContext, operation),
+        getOperationName(dpgContext.program, operation.operation),
+        statusCode
+      );
+      return responseName;
+    });
+  }
+  if (operation.responses && operation.responses.length) {
+    returnTypes.error = getResponseType(
+      operation.responses.filter((r) => isDefaultStatusCode(r.statusCodes))
+    );
+    returnTypes.success = getResponseType(
+      operation.responses.filter((r) => isDefinedStatusCode(r.statusCodes))
+    );
+  }
+  return returnTypes;
+}
+
+/**
+ * Extracts all success or defined status codes for a give operation
+ */
+export function getOperationSuccessStatus(operation: HttpOperation): string[] {
+  const responses = operation.responses ?? [];
+  const status: string[] = [];
+
+  for (const response of responses) {
+    if (isDefinedStatusCode(response.statusCodes)) {
+      status.push(getOperationStatuscode(response));
+    }
+  }
+
+  return status;
+}
+
 export function getOperationStatuscode(
   response: HttpOperationResponse
 ): string {
-  switch (response.statusCode) {
-    case "*":
-      return "default";
-    default:
-      return `${response.statusCode}`;
+  const statusCodes = response.statusCodes;
+  if (statusCodes === "*") {
+    return "default";
+  } else if (typeof statusCodes === "number") {
+    return String(statusCodes);
+  } else {
+    // FIXME - this is a hack to get the first status code
+    // https://github.com/Azure/autorest.typescript/issues/2063
```

Comment:
The status code range support would not be covered in this pr and tracked here: https://github.com/Azure/autorest.typescript/issues/2063.